### ISSUE TITLE
Some Python functions still rely on `inspect.getargspec()` removed in Python 3.11

### ISSUE THREAD

**Issue by krivenko** [TRIQS/triqs#911]
A simple grep on the 3.2.x source directory gives
```
doc/sphinxext/numpydoc/docscrape.py:                argspec = inspect.getargspec(func)
python/triqs/sumk/sumk_discrete.py:            Sigma_Nargs = len(inspect.getargspec(Sigma)[0])
python/triqs/fit/fit.py:        assert len(ins.getargspec(self.function)[0])== len(p00) + 1, "error in number of parameters"
python/triqs/dos/hilbert_transform.py:            assert len(inspect.getargspec(Sigma)[0]) == 1, "Sigma must be a function of a single variable"
```

[`inspect.getargspec()`](https://docs.python.org/3.10/library/inspect.html?highlight=inspect#inspect.getargspec) has been deprecated since version 3.0 and is no more available in 3.11.
[`inspect.getfullargspec()`](https://docs.python.org/3.10/library/inspect.html?highlight=inspect#inspect.getfullargspec) should be used instead.

**Comment by Wentzell** [comment]
Thank you @krivenko for pointing this out!

**Comment by Wentzell** [comment]
Now fixed with 7f0141618
